### ISSUE TITLE
Really really bad M1 Logic Pro Performance

### ISSUE THREAD

**Issue by baconpaul** [surge-synthesizer/surge#5876]
[BadPerf.zip](https://github.com/surge-synthesizer/surge/files/8041621/BadPerf.zip)

With LPX 10.7.2 this just won't even play on an M1. Plays fine on an intel.

https://discussions.apple.com/thread/253438649?answerId=256729786022#256729786022

see discussion there. That user on discord is `FreakyFwoof`

**Comment by baconpaul** [comment]
So a few things I've checked

Monterey 12.0.1 LPX 10.7.2 M1Max

- Build surge as x86 only. Still falls apart.

**Comment by baconpaul** [comment]
Duping here what I put on discord and in the apple forum:

---

Hi so as I mentioned, I'm one of the developers on Surge, and the user who posted here about surge shared their logic file with me. https://github.com/surge-synthesizer/surge/issues/5876

Upon opening the logic project I saw what all of you did, namely just garbage performance. In the case of that project, just completely unplayable. But that performance gap isn't in surge it's somewhere in logic land, I think, especially since I was able to take a set of steps to resolve it. I am curious if these steps help anyone else. (That project has 15 instances of surge some of them using our most CPU intensive features. Once resolved it uses roughly 40% of 4 of my performance cores to play in realtime to the speakers with a buffer size of 64; before resolved it wouldn't even start).

What I did was as follows (I'm on M1Max MBP pretty close to full spec).

Quit logic
Open terminal and type `sudo killall -9 AudioComponentRegistrar`. This will kill and restart the registrar.
Restart logic
Audio/Preferences - pin to 8 threads on 8 performance cores
Logic/Plugin Manager - rescan Surge XT

After those steps it played fine. I then upgraded to os 12.2 and it kept working.

(The reason I tried this is because I did an x86 only build first and that didn't help).

I'm not sure why this fixed the problem (I have a completely half baked theory) but I am curious if these steps help anyone else. If so it may help validate or reject my half baked theory.

If after doing this you have logic projects which use surge and still perform like garbage on lpx10.7/m1max then i would love to know over on that GitHub issue (or here too but I'm not that active on these forums).
